src/globals.c

```diff
@@ -152,34 +152,41 @@ void set_version_url(char *url) {
 bool set_format_string(char *userinput)
 {
 	int version;
+	int ret;
 
-	if (userinput == NULL) {
-		if (format_string) {
-			free(format_string);
+	if (userinput) {
+		// allow "staging" as a format string
+		if ((strcmp(userinput, "staging") == 0)) {
+			if (format_string) {
+				free(format_string);
+			}
+			string_or_die(&format_string, "%s", userinput);
+			return true;
 		}
-		string_or_die(&format_string, "%s", SWUPD_DEFAULT_FORMAT);
-		return true;
-	}
 
-	// allow "staging" as a format string
-	if ((strcmp(userinput, "staging") == 0)) {
+		// otherwise, expect a positive integer
+		errno = 0;
+		version = strtoull(userinput, NULL, 10);
+		if ((errno < 0) || (version <= 0)) {
```

Comment:
The bigger problem is that "version" is an int, not an unsigned type, so I'll make that correction too.